src/Views/CollectionPage.vala

```diff
@@ -324,24 +337,20 @@ public abstract class CollectionPage : MediaPage {
         action.sensitive = sensitive;
     }
 
-    private void populate_external_app_menu (Gtk.Menu menu, PhotoFileFormat file_format, bool raw) {
+    private int populate_external_app_menu (Gtk.Menu menu, string[] mime_types, bool raw) {
         SortedList<AppInfo> external_apps;
-        string[] mime_types;
+        int n_items = 0;
 
         foreach (Gtk.Widget item in menu.get_children ()) {
             menu.remove (item);
         }
 
-        // get list of all applications for the given mime types
-        mime_types = file_format.get_mime_types ();
-
         if (!raw) {
             var files_appinfo = AppInfo.get_default_for_type ("inode/directory", true);
-
-            var files_item_icon = new Gtk.Image.from_gicon (files_appinfo.get_icon (), Gtk.IconSize.MENU);
-            files_item_icon.pixel_size = 16;
-
-            var menuitem_grid = new Gtk.Grid ();
+            var files_item_icon = new Gtk.Image.from_gicon (
+                files_appinfo.get_icon (), Gtk.IconSize.MENU
```

Comment:
When we get app icons from appinfo we need to specify pixel size or Gtk won't resize images, it'll just use whatever the nearest size is. So for example if a file browser were to only supply a 128px icon, that's what Gtk would use at 128px instead of scaling it down to 16px

Reply:
Done